src/ol/renderer/Composite.js

```diff
@@ -98,6 +95,10 @@ class CompositeMapRenderer extends MapRenderer {
       const layerRenderer = this.getLayerRenderer(layer);
       if (layerRenderer.prepareFrame(frameState, layerState)) {
         const element = layerRenderer.renderFrame(frameState, layerState);
+        const zIndex = layerState.zIndex;
+        if (zIndex !== element.style.zIndex) {
```

Comment:
Is this check really necessary? I don't think set an identical `zIndex` will have an effect…

Reply:
My idea is that changing the `zIndex` even if it's the same value will trigger a reflow or something similar (but I haven't tried if it's really the case)

Reply:
I just test to confirm, looks like it isn't the case.